modules/programs/gnome-shell.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.programs.gnome-shell;
+
+  extensionOpts = { config, ... }: {
+    options = {
+      id = mkOption {
+        type = types.str;
+        example = "user-theme@gnome-shell-extensions.gcampax.github.com";
+        description = ''
+          ID of the gnome-shell extension. If not provided, it
+          will be obtained from <varname>package.extensionUuid</varname>.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = "pkgs.gnome3.gnome-shell-extensions";
+        description = ''
+          Package providing a gnome-shell extension in
+          <filename>$out/share/gnome-shell/extensions/''${id}</filename>.
+        '';
+      };
+    };
+
+    config = mkIf (hasAttr "extensionUuid" config.package) {
+      id = mkDefault config.package.extensionUuid;
+    };
+  };
+
+  themeOpts = {
+    options = {
+      name = mkOption {
+        type = types.str;
+        example = "Plata-Noir";
+        description = ''
+          Name of the gnome-shell theme.
+        '';
+      };
+      package = mkOption {
+        type = types.nullOr types.package;
+        default = null;
+        example = literalExpression "pkgs.plata-theme";
+        description = ''
+          Package providing a gnome-shell theme in
+          <filename>$out/share/themes/''${name}/gnome-shell</filename>.
+        '';
+      };
+    };
+  };
```

Comment:
It would be cool if the default value for the theme option was `gtk.theme.{package,name}`.

Having to configure the theme in yet another place in Home Manager is annoying IMO.

Reply:
Good point, I guess you won't need to configure this option though. Personally I have `gtk.theme` configured, but I am not using this `program.gnome-shell.theme`.

I'm guessing that not all Gnome themes need to be GTK themes?

How would one go about to set a default here? I guess it would have to be mkDefault so a user could override it, but it still needs a conditional for the default as it should only be added if `gtk.theme` was configured.

Also `gtk.theme.package` supports being null (if the theme is part of the base offering), so perhaps that would need to be supported for `programs.gnome-shell.theme.package` as well.